pkg/controller/clusterdeprovision/clusterdeprovision_controller.go

```diff
@@ -139,6 +160,12 @@ func (r *ReconcileClusterDeprovision) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
+	// Check if deprovisions are currently disabled: (originates in HiveConfig in real world)
+	if r.deprovisionsDisabled {
+		rLog.Warn("deprovisions are currently disabled in HiveConfig, skipping")
+		return reconcile.Result{}, nil
+	}
```

Comment:
Should we move this under the ClusterDeployment deletion timestamp check so that we get those log messages even when deprovisioning is disabled?

So right above these lines:
```golang
	// Generate an uninstall job
	rLog.Debug("generating uninstall job")
```

I'm just thinking that these might be a good warning / error messages to seee if something mangles the data during the migration.

Reply:
Sounds good to me, updated.